src/libstrongswan/plugins/wolfssl/wolfssl_common.h

```diff
@@ -48,19 +48,22 @@
 #define PARSE_ERROR	WOLFSSL_PARSE_ERROR
 
 /* Remap unused enums from the OpenSSL compatibility layer to avoid conflicts */
-#define ASN1_BOOLEAN         REMAP_ASN1_BOOLEAN
 #define ASN1_OID             REMAP_ASN1_OID
 #define ASN1_INTEGER         REMAP_ASN1_INTEGER
 #define ASN1_BIT_STRING      REMAP_ASN1_BIT_STRING
-#define ASN1_IA5STRING       REMAP_ASN1_IA5STRING
-#define ASN1_OCTET_STRING    REMAP_ASN1_OCTET_STRING
-#define ASN1_UTCTIME         REMAP_ASN1_UTCTIME
-#define ASN1_GENERALIZEDTIME REMAP_ASN1_GENERALIZEDTIME
 
 #ifndef WOLFSSL_USER_SETTINGS
 	#include <wolfssl/options.h>
 #endif
+#include <wolfssl/wolfcrypt/settings.h>
+#ifdef WOLFSSL_PLUGIN_INC_SSL_H
```

Comment:
Why is that wolfssl/ssl.h can't be included?

Reply:
There are several files in the strongSwan wolfSSL plugin which include `wolfssl_common.h,` then proceed to include `<asn1/asn1.h>` file from strongSwan.  The `asn1.h` file includes an enumerated type setting the value of `ASN1_GENERALIZED_TIME` to `0x18`.  Our `wolfssl/ssl.h` file also includes an enumerated type setting `ASN1_GENERALIZED_TIME` to a different value, `4`,  This leads to a compilation error related to redefinition of `ASN1_GENERALIZED_TIME` to different values.  Since these are both enumerated types, we cannot "undef" values to avoid the conflict.

I found three files in the strongSwan wolfSSL plugin which didn't need `ssl.h`, but were including it, and so removing it from those files resolved the conflict.  I also found that a only single file actually needed `ssl.h`,   `wolfssl_plugin.c`, so I chose to create a way to selectively include `ssl.h` to limit changes to files.

I didn't explore this option, but alternatively, if we could remove the enum value `ASN1_GENERALIZED_TIME` from `wolfssl/ssl.h`, that would also resolve the conflict... However, I don't know if it's needed by wolfSSL, or supports some special use-case (?), and by going the other route the changes were confined to the strongSwan wolfSSL plugin... Another alternative would be to change the `wolfssl/ssl.h` definition of `ASN1_GENERALIZED_TIME` value to match that of the `asn1.h` file's... so change to `0x18`, but again, I don't know the ramifications for wolfSSL for making that change...

Reply:
I see. I assume you mean ASN1_GENERALIZEDTIME.
The value doesn't appear to be used and conflicts with our ability to be compatible with OpenSSL as it is the same name as one of its typedefs.
We define ASN1_GENERALIZEDTIME to  WOLFSSL_ASN1_TIME in wolfssl/openssl/ssl.h.
WOLFSSL_ASN1_TIME is a typedef in wolfssl/wolfcrypt/asn_public.h.
I can't see any history on the enum either - it appears to have been added at import time.

Please remove the enum.

Reply:
Yes, I got the name wrong (multiple times!); it's the `ASN1_GENERALIZEDTIME` enum value.

Great!  I'll update this branch to remove the `wolfssl/ssl.h` inclusion-control logic, and open a [PR against wolfSSL](https://github.com/wolfSSL/wolfssl/pull/5667) to remove the `ASN1_GENERALIZETIME` enum value there.  

Thank-you for the insight!

Reply:
The changes to wolfSSL are now merged, so I'm going to start working on upstreaming the changes in this PR.  Please let me know if there are any concerns.  Thanks!